### ISSUE TITLE
failure to bind mount a path with a comma in the name

### ISSUE THREAD

**Issue by sneumann** [apptainer/singularity#5959]
### Version of Singularity:

```
$ singularity version
3.5.2+ds1
```

### Expected behavior

I need to bind-mound a directory into the container that has a comma in the name:
`singularity run -B "foo,bar:/data" docker://alpine ls /data`
Fails with error `mount source /.../bar doesn't exist`, since only the first part of the path is used.

A similar issue is reported in #5923 with colon in the name.

### Actual behavior

```
## Prepare: 
mkdir "foo" "foo,bar" ; touch "foo/bar" "foo,bar/baz"

## Works:
singularity run -B "foo:/data" docker://alpine ls /data

## Fails:
singularity run -B "foo,bar:/data" docker://alpine ls /data
```
with error `mount source /.../bar doesn't exist`. [Docs](https://sylabs.io/guides/3.5/user-guide/bind_paths_and_mounts.html#system-defined-bind-paths) say `-B` takes `a comma-delimited string of bind path specifications`, but I couldn't find how to quote the comma in the filename. I tried `foo\,bar`, `'foo,bar'` but no luck. Ideas ? 

### Steps to reproduce this behavior

See above.

### What OS/distro are you running

```
Ubuntu 18.04.5 LTS
```

### How did you install Singularity

Installed from Debian packages.

**Comment by gmkurtzer** [comment]
Escaping the comma from the shell would not do the same thing once interpreted by Singularity. It might be possible to add an internal escape, but I'll leave decision that up to @cclerget or @ikaneshiro.

**Comment by carterpeel** [comment]
Hello,

This is a templated response that is being sent out to all open issues.  We are working hard on 'rebuilding' the Singularity community, and a major task on the agenda is finding out what issues are still outstanding.


**Please consider the following:**

1. Is this issue a duplicate, or has it been fixed/implemented since being added?
2. Is the issue still relevant to the current state of Singularity's functionality?
3. Would you like to continue discussing this issue or feature request?


Thanks,
Carter

**Comment by sneumann** [comment]
Hi @carterpeel , I am afraid the answers are no, yes and yes. A proper fix will also be a likely fix for #5923 as well, so more than one bird with that stone. Yours, Steffen
